radicle/src/cob/patch.rs

```diff
@@ -376,6 +384,16 @@ impl store::FromHistory for Patch {
                             .into(),
                             op.clock,
                         );
+                        let doc = repo.identity_doc_at(op.identity)?;
+
+                        if revision
+                            .merges()
+                            .filter(|m| doc.is_delegate(&m.node))
+                            .count()
+                            >= doc.threshold
+                        {
+                            self.state.set(State::Merged, op.clock);
+                        }
```

Comment:
How will this threshold work if delegates merge into different levels of the target branch?  For example, what happens if there are two patches, and delegates merge them out of order.

Reply:
Good question! Generally I would say that only one delegate should be doing the merges for that reason. Note that this isn't an issue related to radicle: any model with multiple forks will have to answer this question; and the answer is usually that one person (or a bot) takes care of merging, and the others pull from there.

We could have smarter rules for setting the state to Merged, but it's outside the scope of this PR.